lib/srv/db/mongodb/protocol/opcompressed.go

```diff
@@ -114,6 +114,10 @@ func readOpCompressed(header MessageHeader, payload []byte) (message *MessageOpC
 		CompressedMessage: compressedMessage,
 		bytes:             append(header.bytes[:], payload...),
 	}
+	if uncompressedSize <= 0 || len(compressedMessage) <= 0 {
+		return nil, trace.BadParameter("malformed OP_COMPRESSED: invalid message size %v", payload)
```

Comment:
Should this error message print the size that is invalid, or the actual raw payload?